src/sinks/apex/mod.rs

```diff
@@ -0,0 +1,170 @@
+use bytes::Bytes;
+use futures::FutureExt;
+use futures_util::SinkExt;
+use http::{Request, StatusCode, Uri};
+use hyper::Body;
+use serde_json::json;
+use vector_config::configurable_component;
+
+#[cfg(feature = "apex-integration-tests")]
+#[cfg(test)]
+mod integration_tests;
+
+use crate::{
+    codecs::Transformer,
+    config::{
+        AcknowledgementsConfig, GenerateConfig, Input, SinkConfig, SinkContext, SinkDescription,
+    },
+    event::Event,
+    http::HttpClient,
+    sinks::util::{
+        http::{BatchedHttpSink, HttpEventEncoder, HttpSink},
+        BatchConfig, BoxedRawValue, JsonArrayBuffer, RealtimeSizeBasedDefaultBatchSettings,
+        TowerRequestConfig, UriSerde,
+    },
+};
+
+/// Configuration for the `apex` sink.
+#[configurable_component(sink)]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct ApexSinkConfig {
+    /// The base URI of the Apex instance. Vector will append `/add_events` to this.
+    uri: UriSerde,
+
+    /// The id of the project to associate reported logs with.
+    project_id: String,
+
+    /// The api token to use to authenticate with Apex.
+    api_token: String,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    batch: BatchConfig<RealtimeSizeBasedDefaultBatchSettings>,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    request: TowerRequestConfig,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    acknowledgements: AcknowledgementsConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new::<ApexSinkConfig>("apex")
+}
+
+impl GenerateConfig for ApexSinkConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"uri = "https://your.apex.domain"
+            project_id = "your-apex-project-id"
+            api_token = "your-apex-api-token""#,
+        )
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "apex")]
+impl SinkConfig for ApexSinkConfig {
+    async fn build(
+        &self,
+        cx: SinkContext,
+    ) -> crate::Result<(super::VectorSink, super::Healthcheck)> {
+        let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
+        let batch_settings = self.batch.into_batch_settings()?;
+
+        let buffer = JsonArrayBuffer::new(batch_settings.size);
+
+        let client = HttpClient::new(None, cx.proxy())?;
+
+        let sink = BatchedHttpSink::new(
```

Comment:
We may want to open an issue or drop a comment here to update this to a new-style sink since `BatchedHttpSink` is getting deprecated internally.